pkg/gardenlet/controller/federatedseed/extensions/artifact.go

```diff
@@ -209,6 +230,25 @@ func newStateArtifact(gvk schema.GroupVersionKind, newObjFunc func() client.Obje
 	return a
 }
 
+func newExtensionStatusArtifact(gvk schema.GroupVersionKind, newObjFunc func() client.Object, fn predicateFn) *artifact {
```

Comment:
It's probably not needed to let gardenlet watch all these objects and return the status "asap" similar to the `ShootState`. The `.status.providerStatus` should only change after reconciliation, hence, it would be sufficient if you add another task to the end of the reconciliation flow which syncs the provider statuses, for example.

Reply:
I cannot follow. This is generic functionality that should be used by other components. How do you know that it is not need to "watch all these objects and return the status "asap"? 
Just to name a few possibilities 
 - admission plugins forbid altering of infrastructure resource during Shoot creation,....
 - Third-party that wants to act upon created infrastructure resources. 
   - Why Let them wait until Shoot is reconciled?
   - What happens if the Infrastructure is already created, but the initial creation of the Shoot fails? 
 - During Shoot deletion: Fails before Shoot can be deleted. Infra resources are already gone but still in the `ShootExtensionStatus`.

Anyways, i mentioned a couple of times in our internal syncs that I will implement it this way.
Plus this PR is now a couple of weeks old and this comment would mean to completely change the implementation + tests.
As this functionality has low priority there are much more important things to be tackled, I consider dropping this PR and revert the previous commit with the unused API.
If this becomes a higher priority in the future we can revisit it.

My lesson for this is, that I ll not accept tasks that are supposed to be done as part of a 5-day quality week that will foreseeably take much longer. This is especially true when adding new APIs that always spark discussions. 
In addition, I would appreciate if these kinds of ideas that have a huge impact on the implementation are communicated if possible in advance.

Reply:
IIUC, the implementation leverages the already established watches for syncing the `ShootState` in the `federatedseed` controller, correct? In this case (no new watches are created), it seems to be fine to proceed with the implementation as is. You have enlisted a few cases in which immediate-syncing seems beneficial (assuming the existing watches are used).

In retrospect, even if the change seemed to be small enough, a GEP would've been the proper way for proposing this enhancement. #3873 didn't have much activity (rather none) and there was no "main reviewer" assigned to it. Also, I don't think there was a dedicated meeting about this topic and its implementation details. Probably, it cannot be assumed that there is enough bandwidth to provide such "implementation ideas" ad-hoc just because it was mentioned once or twice in a sync while everybody else is busy with other topics (unless there is a "main reviewer", of course).

Anyways, let's proceed now and finish this up. It's not necessary to revert everything. Significant time was already invested and we still would like to have the feature.